src/azul/service/manifest_service.py

```diff
@@ -815,15 +824,39 @@ def manifest_config(self) -> ManifestConfig:
                               combine_script='return new ArrayList(params._agg.fields)',
                               reduce_script=reduce_script)
         es_search = es_search.extra(size=0)
-        response = es_search.execute()
-        assert len(response.hits) == 0
+        try:
+            response = es_search.execute()
+        # If timeout is exceeded the request is partitioned to reduce load on ES
+        except ConnectionTimeout:
+            fields = self._partition_search(es_search)
+        else:
+            assert len(response.hits) == 0
+            fields = response.aggregations.fields.value
         return {
             'contents': {
                 value: value.split('.')[-1]
-                for value in sorted(response.aggregations.fields.value)
+                for value in sorted(fields)
             }
         }
 
+    def _partition_search(self, es_search: Search) -> Set[str]:
+        requests = [es_search.query(Q('bool',
```

Comment:
there are a lot of `Q`s and `p`s flying around here, I would change the name of `p` to `prefix`.

Reply:
Nit:
There's actually a lot going down in the list comprehension itself, maybe
```
        for prefix in list('0123456789abcdef'):
            kew = Q('bool', must=Q('prefix', **{'bundles.uuid.keyword': prefix}))
            requests.append(es_search.query(kew))
```

Should rename `kew` to something else... 🤔

Reply:
Yea, for `Q` that is the class name of the Elasticsearch query object. However, I added the for loop to increase readability as you suggested.